provider/provider.go

```diff
@@ -45,6 +45,7 @@ type Provider interface {
 	AsOfflineStore() (OfflineStore, error)
 	Type() pt.Type
 	Config() pc.SerializedConfig
+	Check() (bool, error)
```

Comment:
nit: what do you think about just calling it ping?

Reply:
I'm not opposed in theory; however, when it comes to Spark, this "ping" is going to be a lot more involved, in which case `Check` feels more appropriate.

Reply:
`CheckHealth` on the Provider might be better now that I think about it again because the top-level struct doesn't provide context for the operation like it does with `Health.Check(providerName)`.